packages/shared/src/components/post/PostSourceInfo.tsx

```diff
@@ -17,10 +27,35 @@ interface SourceInfoProps {
 function PostSourceInfo({
   date,
   source,
-  size = ProfileImageSize.Medium,
   className,
 }: SourceInfoProps): ReactElement {
+  const isMobile = useViewSize(ViewSize.MobileXL);
+  const [showActionBtn, setShowActionBtn] = useState(false);
   const isUnknown = source.id === 'unknown';
+  const { squad, isLoading: isLoadingSquad } = useSquad({
+    handle: source.handle,
+  });
+  const { data, status } = useContentPreferenceStatusQuery({
+    id: source.id,
+    entity: ContentPreferenceType.Source,
+  });
+
+  useEffect(() => {
+    if (isMobile && status === 'success' && !showActionBtn) {
+      setShowActionBtn(
+        ![
+          ContentPreferenceStatus.Follow,
+          ContentPreferenceStatus.Subscribed,
+        ].includes(data?.status),
+      );
+    }
+  }, [status, data?.status, showActionBtn, isMobile]);
```

Comment:
Wdyt about the useMemo instead?

```suggestion
  const isMobile = useViewSize(ViewSize.MobileXL);
  const isUnknown = source.id === 'unknown';
  const { squad, isLoading: isLoadingSquad } = useSquad({
    handle: source.handle,
  });
  const { data, status } = useContentPreferenceStatusQuery({
    id: source.id,
    entity: ContentPreferenceType.Source,
  });

  const showActionBtn = useMemo(() => {
    if (isMobile && status === 'success' && !showActionBtn) {
      return ![
            ContentPreferenceStatus.Follow,
            ContentPreferenceStatus.Subscribed,
          ].includes(data?.status);
    }

    return false;
  }, [status, data?.status, showActionBtn, isMobile]);
  ```

Reply:
Oh. Right, overlooked that. The thing is, it is not ideal to be using state and use effect solely to update a state. There can be many ways to avoid the use effect, but your call.